storage/src/persistent/commit_log/mod.rs

```diff
@@ -16,23 +15,185 @@ use crate::persistent::schema::{CommitLogDescriptor, CommitLogSchema};
 use crate::persistent::BincodeEncoded;
 
 pub type CommitLogRef = Arc<RwLock<CommitLog>>;
+pub mod error;
+mod reader;
+mod writer;
+
+const INDEX_FILE_NAME: &str = "table.index";
+const DATA_FILE_NAME: &str = "table.data";
+
+const TH_LENGTH: usize = 24;
+
+pub type Message = Vec<u8>;
+pub struct MessageSet {
+    cursor: usize,
+    indexes: Vec<Index>,
+    acc: usize,
+    buf: Vec<u8>,
+}
+
+impl MessageSet {
+    fn new(indexes: Vec<Index>, buf: Vec<u8>) -> Self {
+        Self {
+            cursor: 0,
+            indexes,
+            acc: 0,
+            buf,
+        }
+    }
+}
+
+impl Iterator for MessageSet {
+    type Item = Message;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let index = self.indexes.get(self.cursor)?;
+        let data = &self.buf[self.acc..(self.acc + index.uncompressed_data_length as usize)];
+        self.acc += index.uncompressed_data_length as usize;
+        self.cursor += 1;
+        Some(data.to_vec())
```

Comment:
I don't know how performance sensitive is this code, so maybe this doesn't matter: we are allocating a new vector each time here, is that necessary?

Reply:
Found https://docs.rs/streaming-iterator/ which would solve the issue.

But as I said, if performance is not an issue here, then it doesn't matter.